packages/zosuss/src/api/Shell.ts

```
@@ -0,0 +1,73 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Logger } from "@brightside/imperative";
+import { Client, ClientChannel } from "ssh2";
+import { SshSession } from "../SshSession";
+
+export class Shell {
+    public static executeSsh(session: SshSession, command: string, callback: any): void {
+        const conn = new Client();
+        let privateKeyPath;
+        try {
+            privateKeyPath = require("fs").readFileSync(session.ISshSession.privateKey);
+        } catch (err) {
+            privateKeyPath = "";
```

I think we should return an error message here to tell the user we couldn't read the private key.  Do we have any tests that cover this case?

---

This was for checking privateKey being empty. When it is empty, readFileSync() returns error.
I've changed it to make it clearer:

`       let privateKeyPath;
        try {
            privateKeyPath = require("fs").readFileSync(session.ISshSession.privateKey);
        } catch (err) {
            if (err.message.includes("Received type undefined")) {
                privateKeyPath = "";
            }
        }`

---

It was for checking privateKey being empty. When it was empty, readFileSync() failed and gave an error.
We don't have any tests for the publickey authentication feature at the moment.

---

Ah I see. So could you if privateKey is empty first rather than triggering the error?

---

I tried with that, but readFileSync() seems to be running no matter.

---

I tried it and didn't work last time, but I figured this time that when it's not defined on the profile, it is stored as "undefined" literally as string type rather than object type undefined.